backend/remote-state/http/client.go

```diff
@@ -0,0 +1,298 @@
+	case http.StatusOK:
+		return nil
+	default:
+		return fmt.Errorf("HTTP error: %d", resp.StatusCode)
+	}
+}
+
+// Lock writes to a lock file, ensuring file creation. Returns the generation number, which must be passed to Unlock().
+func (c *RemoteClient) Lock(info *state.LockInfo) (string, error) {
+
+	lockURL, _ := url.Parse(c.lockAddress)
+	c.lockID = ""
+
+	jsonLockInfo := info.Marshal()
+
+	resp, err := c.httpRequest(c.lockMethod, lockURL, &jsonLockInfo, "lock")
+	if err != nil {
+		return "", err
+	}
+
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		c.lockID = info.ID
+		c.jsonLockInfo = jsonLockInfo
+		return info.ID, nil
+	case http.StatusConflict, http.StatusLocked:
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+
+		if err != nil {
+			return "", fmt.Errorf("HTTP remote state already locked, failed to read body")
+		}
+		existing := state.LockInfo{}
+		err = json.Unmarshal(body, &existing)
+		if err != nil {
+			return "", fmt.Errorf("HTTP remote state already locked, failed to unmarshal body")
+		}
+		return "", c.lockError(fmt.Errorf("HTTP remote state already locked: ID=%s", existing.ID))
+	default:
+		return "", fmt.Errorf("Unexpected HTTP response code %d", resp.StatusCode)
+	}
+}
+
+// Unlock the state with id from lock.
+func (c *RemoteClient) Unlock(id string) error {
+
+	lockErr := &state.LockError{}
+
+	lockInfo, err := c.lockInfo()
+	if err != nil {
+		lockErr.Err = fmt.Errorf("failed to retrieve lock info: %s", err)
+		return lockErr
+	}
+	lockErr.Info = lockInfo
+
+	if lockInfo.ID != id {
+		lockErr.Err = fmt.Errorf("lock id %q does not match existing lock", id)
+		return lockErr
+	}
+
+	unlockURL, _ := url.Parse(c.unlockAddress)
+	resp, err := c.httpRequest(c.unlockMethod, unlockURL, &c.jsonLockInfo, "unlock")
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		return nil
+	default:
+		return fmt.Errorf("Unexpected HTTP response code %d", resp.StatusCode)
+	}
+}
+
+func (c *RemoteClient) lockError(err error) *state.LockError {
+	lockErr := &state.LockError{
+		Err: err,
+	}
+
+	info, infoErr := c.lockInfo()
+	if infoErr != nil {
+		lockErr.Err = multierror.Append(lockErr.Err, infoErr)
+	} else {
+		lockErr.Info = info
+	}
+	return lockErr
+}
+
+// lockInfo reads the lock file, parses its contents and returns the parsed
+// LockInfo struct.
+func (c *RemoteClient) lockInfo() (*state.LockInfo, error) {
+	// Convert address to type URL
+	lockURL, _ := url.Parse(c.lockAddress)
+
+	resp, err := c.httpRequest("GET", lockURL, nil, "get lock")
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+
+	// Read in the body
+	buf := bytes.NewBuffer(nil)
+	if _, err := io.Copy(buf, resp.Body); err != nil {
+		return nil, fmt.Errorf("Failed to read lock file: %s", err)
+	}
+
+	info := &state.LockInfo{}
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		readBuf, _ := ioutil.ReadAll(buf)
```

Comment:
why do you suppress error here?

Reply:
don't think is needed since if readBuf has an error allocating memory I think there is a bigger problem on the system? but...why not. :)